src/GraphODataTemplateWriter/CodeHelpers/Java/TypeHelperJava.cs

```diff
@@ -944,10 +890,28 @@ public static string CreatePackageDefForEntity(this CustomT4Host host)
                 }
             }
 
+            var collidedProperties = new HashSet<string>();
+            var allProperties = new HashSet<string>();
+            foreach(var property in properties)
+            {
+                var rawTypeName = property.Projection.Type.Name;
+                if (allProperties.Contains(rawTypeName))
+                {
+                    collidedProperties.Add(rawTypeName);
+                }
+                allProperties.Add(rawTypeName);
```

Comment:
else block

Reply:
> Thanks for taking this on, please also revert the vipr submodule change, I believe your submodule was behind when you committed. (git submodule update --remote --merge) For some reason I see a lot of whitespace changes in the PR you put together here? [microsoftgraph/msgraph-beta-sdk-java#351](https://github.com/microsoftgraph/msgraph-beta-sdk-java/pull/351)
Also seeing the whitespace, I believe this may be an issue on the github side. If we look at the line numbers, they still match up for files which are unchanged and off by a couple for those which have a few line changes. Not sure why the empty space is being shown.